Broken Dependencies with `apoc.mongo`

## Guidelines

Please note that GitHub issues are only meant for bug reports/feature requests. If you have questions on how to use APOC, please ask on [the Neo4j Discussion Forum](https://community.neo4j.com/c/neo4j-graph-platform/procedures-apoc) instead of creating an issue here.

## Expected Behavior (Mandatory)
The following APOC procedures fail with dependency issues:

```shell
call apoc.mongo.count('mongodb://user:pass@mongo:27017/database.collection?authSource=admin', {}) yield value return value
```

results in 
```shell
Failed to invoke procedure `apoc.mongo.count`: Caused by: java.lang.ClassNotFoundException: com.mongodb.client.MongoClients
```

## Actual Behavior (Mandatory)
I would expect this to return the count of documents, but it seems that possibly the mongoDB driver wasn't updated appropriately.

## How to Reproduce the Problem

### Simple Dataset (where it's possibile)

```
//Insert here a set of Cypher statements that helps us to reproduce the problem

```


### Steps (Mandatory)

  1. I installed Neo4j via docker and installed the plugins required:
 ```shell
$ ls
apoc-5.19.0-core.jar                            apoc-5.19.0-extended.jar                        apoc-mongodb-dependencies-5.19.0-all.jar
 ``` 
  1. I think ran the above APOC procedure:
```shell
call apoc.mongo.count('mongodb://user:pass@mongo:27017/database.collection?authSource=admin', {}) yield value return value
```

## Screenshots (where it's possibile)
![Screenshot 2024-04-21 at 9 39 11 AM](https://github.com/neo4j-contrib/neo4j-apoc-procedures/assets/54959686/7c5cfc7f-6535-4ca2-a107-11459022073a)

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: `Darwin ARM64` (Docker)
  - Neo4j: `15.19-community` (Docker)
  - Neo4j-Apoc: `5.19.0`

--- a/extra-dependencies/mongodb/build.gradle
+++ b/extra-dependencies/mongodb/build.gradle
@@ -17,7 +17,7 @@ jar {
 }
 
 dependencies {
-    implementation 'org.mongodb:mongodb-driver:3.2.2', commonExclusions
+    implementation 'org.mongodb:mongodb-driver-sync:4.11.1', commonExclusions
 }
 
 

